.github/workflows/multicluster.yaml

```diff
@@ -0,0 +1,245 @@
+  #    - "labeled"
+  ###
+
+env:
+  clusterName1: cilium-cli-ci-${{ github.run_id }}-multicluster-1
+  clusterName2: cilium-cli-ci-${{ github.run_id }}-multicluster-2
+  zone: us-west2-a
+  check_url: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}
+
+jobs:
+  installation-and-connectivitiy:
+    if: |
+      (github.event.issue.pull_request && startsWith(github.event.comment.body, 'ci-multicluster')) ||
+      github.event_name == 'push' ||
+      github.event.label.name == 'ci-run/multicluster'
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    steps:
+      - name: Set up job variables
+        id: vars
+        run: |
+          if [ ${{ github.event.issue.pull_request || github.event.pull_request }} ]; then
+            PR_API_JSON=$(curl \
+              -H "Accept: application/vnd.github.v3+json" \
+              -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+              ${{ github.event.issue.pull_request.url || github.event.pull_request.url }})
+            PR_SHA=$(echo "$PR_API_JSON" | jq -r ".head.sha")
+            PR_NUMBER=$(echo "$PR_API_JSON" | jq -r ".number")
+            echo ::set-output name=is_pr::true
+            echo ::set-output name=sha::${PR_SHA}
+            echo ::set-output name=owner::${PR_NUMBER}
+          else
+            echo ::set-output name=is_pr::
+            echo ::set-output name=sha::${{ github.sha }}
+            echo ::set-output name=owner::${{ github.sha }}
+          fi
+
+      - name: Set PR status check to pending
+        if: ${{ steps.vars.outputs.is_pr }}
+        uses: Sibz/github-status-action@67af1f4042a5a790681aad83c44008ca6cfab83d
+        with:
+          authToken: ${{ secrets.GITHUB_TOKEN }}
+          sha: ${{ steps.vars.outputs.sha }}
+          context: ${{ github.workflow }}
+          description: Connectivity test in progress...
+          state: pending
+          target_url: ${{ env.check_url }}
+
+      - name: Install Cilium CLI
+        run: |
+          curl -LO https://github.com/cilium/cilium-cli/releases/download/v0.4/cilium-linux-amd64.tar.gz
+          sudo tar xzvfC cilium-linux-amd64.tar.gz /usr/bin
+          rm cilium-linux-amd64.tar.gz
+
+      - name: Set up gcloud CLI
+        uses: google-github-actions/setup-gcloud@daadedc81d5f9d3c06d2c92f49202a3cc2b919ba
+        with:
+          project_id: ${{ secrets.GCP_PROJECT_ID }}
+          service_account_key: ${{ secrets.GCP_PR_SA_KEY }}
+          export_default_credentials: true
+
+      - name: Display gcloud CLI info
+        run: |
+          gcloud info
+
+      - name: Create GKE cluster 1
+        run: |
+          gcloud container clusters create ${{ env.clusterName1 }} \
+            --labels "usage=pr,owner=${{ steps.vars.outputs.owner }}" \
+            --zone ${{ env.zone }} \
+            --preemptible \
+            --image-type COS \
+            --num-nodes 2 \
+            --machine-type n1-standard-4
+
+      - name: Create GKE cluster 2
```

Comment:
We could potentially parallelize this by splitting this workflow into separate jobs that will create the clusters, wait for images and make the final job require these three jobs to run.

Reply:
Yea, I was thinking about that as well! I intend to do it at the same time we refactor the other workflows to use matrices, as per the discussion [here](https://cilium.slack.com/archives/C7PE7V806/p1618406776180900).